go-controller/pkg/ovn/namespace.go

```diff
@@ -330,26 +330,35 @@ func (oc *DefaultNetworkController) getAllHostNamespaceAddresses() []net.IP {
 				// skip hybrid overlay nodes
 				continue
 			}
-			hostSubnets, err := util.ParseNodeHostSubnetAnnotation(node, types.DefaultNetworkName)
+			hostNetworkIPs, err := oc.getHostNamespaceAddressesForNode(node)
 			if err != nil {
-				klog.Warningf("Error parsing host subnet annotation for node %s (%v)",
-					node.Name, err)
-			}
-			for _, hostSubnet := range hostSubnets {
-				mgmtIfAddr := util.GetNodeManagementIfAddr(hostSubnet)
-				ips = append(ips, mgmtIfAddr.IP)
-			}
-			// for shared gateway mode we will use LRP IPs to SNAT host network traffic
-			// so add these to the address set.
-			lrpIPs, err := util.ParseNodeGatewayRouterLRPAddrs(node)
-			if err != nil {
-				klog.Errorf("Failed to get join switch port IP address for node %s: %v", node.Name, err)
-			}
-
-			for _, lrpIP := range lrpIPs {
-				ips = append(ips, lrpIP.IP)
+				klog.Errorf("Error parsing annotation for node %s: %v", node.Name, err)
 			}
+			ips = append(ips, hostNetworkIPs...)
 		}
 	}
 	return ips
 }
+
+func (oc *DefaultNetworkController) getHostNamespaceAddressesForNode(node *kapi.Node) ([]net.IP, error) {
```

Comment:
nit: add function description for this function

Reply:
Ack